### PR TITLE
Luciano

### PR DESCRIPTION
**Pull request by lucianonooijen**

### CODE REVIEW

File: models/team.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            .insert(newTeam)
+            .returning(returning);
+        const addedTeam: ITeam = addedTeamArr[0];
+        const team = new Team(addedTeam);
+        return team;
+    }
+
+    public static async getTeamById(id: number): Promise<Team> {
+        const teamArr = await knex.select('*').from('teams').where({ id });
+        const teamData: ITeam = teamArr[0];
```
Comment (JeroenVanSteijn) [inline models/team.ts:52]: You could use knex's .first() method instead of doing teamArr[0]

File: models/team.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import knex from '../helpers/db';
+import Player from './player';
+import ITeam from '../typescript/ITeam';
+import ITeamNew from '../typescript/ITeamNew';
+
+class Team implements ITeamClass {
+    private id: number;
+    private name: string;
+    private invitecode?: string;
```
Comment (JeroenVanSteijn) [inline models/team.ts:9]: Why the ? behind invitecode?

File: models/player.ts [hunk 3]
```diff
@@ -82,22 +102,12 @@ const getNewPlayerData = async (newPlayer: IPlayerNew): Promise<IPlayerNew> => {
...
-    timecreated: Date;
-}
-
-interface IPlayerNew {
-    username: string;
-    password: string;
-    email: string;
-    teamid: number;
-}
+const getRandomUsername = (): String => {
```
Comment (JeroenVanSteijn) [inline models/player.ts:105]: Maybe add a todo that checks if this username exists, although the chances are pretty slim.

File: models/player.ts [hunk 4]
```diff
@@ -64,12 +66,30 @@ class Player implements IPlayerClass {
...
+        const newPlayerData = {
+            username: newUsername,
+            password: `${newUsername}-pass`,
+            email: `${newUsername}@pingpong.bytecode.nl`
+        };
+        const updatedUser = { ...playerCurrent, ...newPlayerData };
+        const deletedPlayerArr = await knex('players')
+            .returning([ 'id' ])
+            .where({ id: userId })
+            .update(updatedUser);
```
Comment (JeroenVanSteijn) [inline models/player.ts:88]: You could use .first() again

File: models/player.ts (hunk 3 above)
Comment (lucianonooijen) [reply models/player.ts:105]: Meh

**Pull request merged**